pvmismatch/pvmismatch_lib/pvmodule.py

```diff
@@ -194,12 +226,19 @@ def __init__(self, cell_pos=STD96, pvcells=None, pvconst=None,
                 if p.pvconst is not pvconst:
                     raise Exception('PVconstant must be the same for all cells')
         self.pvconst = pvconst  #: configuration constants
-        self.Vbypass = Vbypass  #: [V] trigger voltage of bypass diode
+        
+        # set default value of Vbypass if None
+        if Vbypass is None:
+            self.Vbypass = VBYPASS  #: [V] trigger voltage of bypass diode
+        else:
+            # if an object is passed, use that to determine the config of bypass diodes
+            self.Vbypass = Vbypass
+        self.Vbypass_config = parse_diode_config(self.Vbypass, self.cell_pos)
+
         self.cellArea = cellArea  #: [cm^2] cell area
         # check cell position pattern matches list of cells
         if len(pvcells) != self.numberCells:
-            # TODO: use pvexception
-            raise Exception(
+            raise PVexception(
```

Comment:
two suggestions:
1. I believe a [`ValueError`](https://docs.python.org/3/library/exceptions.html#ValueError) would be better here
    >Raised when an operation or function receives an argument that has the right type but an inappropriate value ...
2. [`PVexception` is the base class from which all pvmismatch exceptions are derived](https://github.com/SunPower/PVMismatch/blob/master/pvmismatch/pvmismatch_lib/pvexceptions.py#L8-L12), so if you want to use this, then you should create a new `PVexception`, _eg_ `BypassDiodeConfigError` in `pvexceptions.py`.

Reply:
@chetan201 PTAL